cylc/flow/prerequisite.py

```diff
@@ -79,6 +79,25 @@ def __init__(self, point, start_point=None):
         # * `False` (prerequisite unsatisfied).
         self._all_satisfied = None
 
+    def __str__(self):
```

Comment:
This will help with prerequisite sorting.

There are two places in the data store where we iterate over prereqs:

```python
for prereq in itask.state.prerequisites:
```

With this __str__ method, if we did this:

```python
for prereq in sorted(itask.state.prerequisites, key=str):
```

They should sort ok. Be aware we build this by default, not just on request though so efficiency will matter.

Reply:
I've gone for sorting once at task proxy state initialization. The order is preserved by the data store.

Reply:
This is now off-topic for this branch. I'll do a separate follow-up for sorting.